codec/src/main/java/io/netty/handler/codec/compression/JdkZlibEncoder.java

```diff
@@ -34,6 +35,10 @@
  * Compresses a {@link ByteBuf} using the deflate algorithm.
  */
 public class JdkZlibEncoder extends ZlibEncoder {
+    /**
+     * Max size for temporary heap buffers used to copy input data to heap.
+     */
+    static final int MAX_TEMP_HEAP_BUFFER_SIZE = 65536;
```

Comment:
This should actually be configurable via System Property.

Reply:
Should it be a system property, a builder setting, or both? What do you think.

Reply:
A builder or anything direct-based configuration is not a feasible solution because this handler is internally used for various operations.

So a system property should be sufficient.